cuebot/src/main/java/com/imageworks/spcue/util/FrameSet.java

```diff
@@ -44,6 +48,140 @@ public int index(int idx) {
         return frameList.indexOf(idx);
     }
 
+    /**
+     * Return a string representation of a subset of a frame range.
+     *
+     * This approach was adapted from https://pypi.org/project/Fileseq/
+     * @param startFrame Start frame
+     * @param endFrame   End frame
+     * @param stride     The step between frames
+     * @param zFill      Left pad each frame ID with zeroes to make it this width.
+     * @return           String representation of the frame range, e.g. 1-1001x3
+     */
+    private String buildFrangePart(int startFrame, int endFrame, int stride, int zFill) {
+        String padStart = pad(startFrame, zFill);
+        String padStop  = pad(endFrame, zFill);
+        if (startFrame == endFrame) {
+            return padStart;
+        } else if (stride == 1) {
+            return String.format("%s-%s", padStart, padStop);
+        } else {
+            return String.format("%s-%sx%d", padStart, padStop, stride);
+        }
+    }
+
+    /**
+     * Left pad 'number' with zeroes to make it 'width' characters wide.
+     * @param number An integer to pad
+     * @param width  The required width of the resulting String.
+     * @return       A left-padded string, e.g 0047
+     */
+    private String pad(int number, int width){
+        return Strings.padStart(String.valueOf(number), width, '0');
+    }
+
+    /**
+     * Return a String representation of a frame range based on a list of literal integer frame IDs.
+     * @param frames  A list of integers representing frame IDs,
+     * @return        A representation of a frameset, e.g. '1-10,12-100x2'
+     */
+    private String framesToFrameRanges(ImmutableList frames) {
+        return framesToFrameRanges(frames, 0);
+    }
+
+    /**
+     * Return a String representation of a frame range based on a list of literal integer frame IDs.
+     * @param frames  A list of integers representing frame IDs,
+     * @param zFill   The required width of each left-padded frame ID, e.g. 4 if ID '1' should become '0001'
+     * @return        A representation of a frameset, e.g. '1-10,12-100x2'
+     */
+    private String framesToFrameRanges(ImmutableList<Integer> frames, int zFill) {
```

Comment:
It doesn't look like there's a way to use `zFill` at the moment, is there? Since these are all private methods, and the public methods don't have a corresponding argument. Should we just drop it?

If you think it's worth keeping it around, let's add tests that utilize it.

Reply:
I removed all the `zFill` and `pad` method stuff that's not really being used.